src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -13,6 +13,7 @@ public class PublishAssembliesResolver
     {
         private readonly IPackageResolver _packageResolver;
         private IEnumerable<string> _privateAssetPackageIds;
+        private bool _preserveCacheLayout;
```

Comment:
Can you add a unit test or two for the new behavior?

Along with that idea - it would be good to add an integration test or two in [https://github.com/dotnet/sdk/blob/master/test/Microsoft.NET.Publish.Tests/](https://github.com/dotnet/sdk/blob/master/test/Microsoft.NET.Publish.Tests/).